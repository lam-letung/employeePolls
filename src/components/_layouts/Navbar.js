import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { handleLogoutAuthedUser } from "../../redux/actions/authedUser";
import { FaHome, FaPollH, FaTrophy, FaSignOutAlt, FaSignInAlt } from 'react-icons/fa';

const Navbar = ({ dispatch, authUser }) => {
  const logoutHandle = () => {
    dispatch(handleLogoutAuthedUser());
  };

  return (
    <nav className="navbar bg-gradient-to-r from-blue-500 to-indigo-600 p-4 shadow-lg text-white">
      <div className="flex-1">
        <Link data-testid="logo" to="/" className="btn btn-ghost normal-case text-2xl text-white font-bold">
          Employee Polls
        </Link>
      </div>
      <div className="flex-none mx-auto">
        {authUser && (
          <h2 data-testid="authUserName" className="text-2xl font-semibold">
            {authUser?.name}
          </h2>
        )}
      </div>
      <div className="flex-none">
        <ul data-testid="navbuttons" className="menu menu-horizontal space-x-4">
          <li>
            <Link data-testid="home-link" to="/" className="flex items-center space-x-2 hover:text-gray-200">
              <FaHome /> <span>Home</span>
            </Link>
          </li>
          <li>
            <Link data-testid="new-poll-link" to="/add" className="flex items-center space-x-2 hover:text-gray-200">
              <FaPollH /> <span>New Poll</span>
            </Link>
          </li>
          <li>
            <Link data-testid="leaderboard-link" to="/leaderboard" className="flex items-center space-x-2 hover:text-gray-200">
              <FaTrophy /> <span>Leaderboard</span>
            </Link>
          </li>
          {authUser ? (
            <li>
              <button data-testid="logout-link" onClick={logoutHandle} className="flex items-center space-x-2 hover:text-gray-200">
                <FaSignOutAlt /> <span>Logout</span>
              </button>
            </li>
          ) : (
            <li>
              <Link data-testid="login-link" to="/login" className="flex items-center space-x-2 hover:text-gray-200">
                <FaSignInAlt /> <span>Login</span>
              </Link>
            </li>
          )}
        </ul>
      </div>
    </nav>
  );
};

const mapStateToProps = ({ authUser }) => ({
  authUser,
});

export default connect(mapStateToProps)(Navbar);
