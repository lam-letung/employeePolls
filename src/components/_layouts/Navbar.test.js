import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import React from "react";
import Navbar from "../_layouts/Navbar";
import { configureStore } from "@reduxjs/toolkit";
import reducer from "../../redux/reducers";
import { loginAuthedUser } from "../../redux/actions/authedUser";
import '@testing-library/jest-dom';


const store = configureStore({ reducer });

describe("Navbar", () => {
  beforeEach(() => {
    store.dispatch(loginAuthedUser({
      username: "tylermcginnis", password: ""
    }));
  });

  test("should render the component", () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>
    );
    expect(view).toBeDefined();
    expect(view).toMatchSnapshot();
  });

  test("should display all elements", () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>
    );

    const homeLinkElement = screen.getByTestId("home-link");
    const newPollLinkElement = screen.getByTestId("new-poll-link");
    const leaderboardLinkElement = screen.getByTestId("leaderboard-link");
    const logoutLinkElement = screen.getByTestId("logout-link");

    expect(homeLinkElement).toHaveTextContent("Home");
    expect(newPollLinkElement).toHaveTextContent("New Poll");
    expect(leaderboardLinkElement).toHaveTextContent("Leaderboard");
    expect(logoutLinkElement).toHaveTextContent("Logout");
  });
});
