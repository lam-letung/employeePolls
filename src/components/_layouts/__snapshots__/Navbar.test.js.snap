// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar bg-gradient-to-r from-blue-500 to-indigo-600 p-4 shadow-lg text-white"
      >
        <div
          class="flex-1"
        >
          <a
            class="btn btn-ghost normal-case text-2xl text-white font-bold"
            data-testid="logo"
            href="/"
          >
            Employee Polls
          </a>
        </div>
        <div
          class="flex-none mx-auto"
        >
          <h2
            class="text-2xl font-semibold"
            data-testid="authUserName"
          />
        </div>
        <div
          class="flex-none"
        >
          <ul
            class="menu menu-horizontal space-x-4"
            data-testid="navbuttons"
          >
            <li>
              <a
                class="flex items-center space-x-2 hover:text-gray-200"
                data-testid="home-link"
                href="/"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
                  />
                </svg>
                 
                <span>
                  Home
                </span>
              </a>
            </li>
            <li>
              <a
                class="flex items-center space-x-2 hover:text-gray-200"
                data-testid="new-poll-link"
                href="/add"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M448 432V80c0-26.5-21.5-48-48-48H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48zM112 192c-8.84 0-16-7.16-16-16v-32c0-8.84 7.16-16 16-16h128c8.84 0 16 7.16 16 16v32c0 8.84-7.16 16-16 16H112zm0 96c-8.84 0-16-7.16-16-16v-32c0-8.84 7.16-16 16-16h224c8.84 0 16 7.16 16 16v32c0 8.84-7.16 16-16 16H112zm0 96c-8.84 0-16-7.16-16-16v-32c0-8.84 7.16-16 16-16h64c8.84 0 16 7.16 16 16v32c0 8.84-7.16 16-16 16h-64z"
                  />
                </svg>
                 
                <span>
                  New Poll
                </span>
              </a>
            </li>
            <li>
              <a
                class="flex items-center space-x-2 hover:text-gray-200"
                data-testid="leaderboard-link"
                href="/leaderboard"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M552 64H448V24c0-13.3-10.7-24-24-24H152c-13.3 0-24 10.7-24 24v40H24C10.7 64 0 74.7 0 88v56c0 35.7 22.5 72.4 61.9 100.7 31.5 22.7 69.8 37.1 110 41.7C203.3 338.5 240 360 240 360v72h-48c-35.3 0-64 20.7-64 56v12c0 6.6 5.4 12 12 12h296c6.6 0 12-5.4 12-12v-12c0-35.3-28.7-56-64-56h-48v-72s36.7-21.5 68.1-73.6c40.3-4.6 78.6-19 110-41.7 39.3-28.3 61.9-65 61.9-100.7V88c0-13.3-10.7-24-24-24zM99.3 192.8C74.9 175.2 64 155.6 64 144v-16h64.2c1 32.6 5.8 61.2 12.8 86.2-15.1-5.2-29.2-12.4-41.7-21.4zM512 144c0 16.1-17.7 36.1-35.3 48.8-12.5 9-26.7 16.2-41.8 21.4 7-25 11.8-53.6 12.8-86.2H512v16z"
                  />
                </svg>
                 
                <span>
                  Leaderboard
                </span>
              </a>
            </li>
            <li>
              <button
                class="flex items-center space-x-2 hover:text-gray-200"
                data-testid="logout-link"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
                  />
                </svg>
                 
                <span>
                  Logout
                </span>
              </button>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar bg-gradient-to-r from-blue-500 to-indigo-600 p-4 shadow-lg text-white"
    >
      <div
        class="flex-1"
      >
        <a
          class="btn btn-ghost normal-case text-2xl text-white font-bold"
          data-testid="logo"
          href="/"
        >
          Employee Polls
        </a>
      </div>
      <div
        class="flex-none mx-auto"
      >
        <h2
          class="text-2xl font-semibold"
          data-testid="authUserName"
        />
      </div>
      <div
        class="flex-none"
      >
        <ul
          class="menu menu-horizontal space-x-4"
          data-testid="navbuttons"
        >
          <li>
            <a
              class="flex items-center space-x-2 hover:text-gray-200"
              data-testid="home-link"
              href="/"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
                />
              </svg>
               
              <span>
                Home
              </span>
            </a>
          </li>
          <li>
            <a
              class="flex items-center space-x-2 hover:text-gray-200"
              data-testid="new-poll-link"
              href="/add"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M448 432V80c0-26.5-21.5-48-48-48H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48zM112 192c-8.84 0-16-7.16-16-16v-32c0-8.84 7.16-16 16-16h128c8.84 0 16 7.16 16 16v32c0 8.84-7.16 16-16 16H112zm0 96c-8.84 0-16-7.16-16-16v-32c0-8.84 7.16-16 16-16h224c8.84 0 16 7.16 16 16v32c0 8.84-7.16 16-16 16H112zm0 96c-8.84 0-16-7.16-16-16v-32c0-8.84 7.16-16 16-16h64c8.84 0 16 7.16 16 16v32c0 8.84-7.16 16-16 16h-64z"
                />
              </svg>
               
              <span>
                New Poll
              </span>
            </a>
          </li>
          <li>
            <a
              class="flex items-center space-x-2 hover:text-gray-200"
              data-testid="leaderboard-link"
              href="/leaderboard"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M552 64H448V24c0-13.3-10.7-24-24-24H152c-13.3 0-24 10.7-24 24v40H24C10.7 64 0 74.7 0 88v56c0 35.7 22.5 72.4 61.9 100.7 31.5 22.7 69.8 37.1 110 41.7C203.3 338.5 240 360 240 360v72h-48c-35.3 0-64 20.7-64 56v12c0 6.6 5.4 12 12 12h296c6.6 0 12-5.4 12-12v-12c0-35.3-28.7-56-64-56h-48v-72s36.7-21.5 68.1-73.6c40.3-4.6 78.6-19 110-41.7 39.3-28.3 61.9-65 61.9-100.7V88c0-13.3-10.7-24-24-24zM99.3 192.8C74.9 175.2 64 155.6 64 144v-16h64.2c1 32.6 5.8 61.2 12.8 86.2-15.1-5.2-29.2-12.4-41.7-21.4zM512 144c0 16.1-17.7 36.1-35.3 48.8-12.5 9-26.7 16.2-41.8 21.4 7-25 11.8-53.6 12.8-86.2H512v16z"
                />
              </svg>
               
              <span>
                Leaderboard
              </span>
            </a>
          </li>
          <li>
            <button
              class="flex items-center space-x-2 hover:text-gray-200"
              data-testid="logout-link"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
                />
              </svg>
               
              <span>
                Logout
              </span>
            </button>
          </li>
        </ul>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
